src/mainboard/seeed/beaglev/src/main.rs

```diff
@@ -0,0 +1,122 @@
+#![feature(llvm_asm)]
+#![feature(lang_items, start)]
+#![no_std]
+#![no_main]
+#![feature(global_asm)]
+#![deny(warnings)]
+
+//use core::fmt::Write;
+use core::intrinsics::transmute;
+use core::panic::PanicInfo;
+use core::sync::atomic::{spin_loop_hint, AtomicUsize, Ordering};
+use model::Driver;
+use payloads::payload;
+use soc::clock::Clock;
+//use soc::syscon::Syscon;
+//use soc::iopad::IOpad;
+use soc::iopadctl::IOpadctl;
+use soc::rstgen::RSTgen;
+//use uart::sifive::SiFive;
+pub mod uart;
+
+global_asm!(include_str!(
+    "../../../../../src/soc/starfive/jh7100/src/bootblock.S"
+));
+global_asm!(include_str!(
+    "../../../../../src/soc/starfive/jh7100/src/init.S"
+));
+
+// TODO: For some reason, on hardware, a1 is not the address of the dtb, so we hard-code the device
```

Comment:
Do we expect the hardware to pass a DT to us?

Reply:
the bootblock passes it and on the beagle there is a vic_secondboot that *may* pass it. Still not sure.